tracing-fmt/src/writer.rs

```diff
@@ -0,0 +1,154 @@
+//! Abstractions for creating [`io::Write`] instances.
+//!
+//! [`io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+
+use std::io;
+
+/// A type that can create [`io::Write`] instances.
+///
+/// `MakeWriter` is used by [`FmtSubscriber`] to print formatted text representations of
+/// [`Event`]s.
+///
+/// This trait is already implemented for function pointers and immutably-borrowing closures that
+/// return an instance of [`io::Write`], such as [`io::stdout`] and [`io::stderr`].
+///
+/// [`io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+/// [`FmtSubscriber`]: struct.FmtSubscriber.html
```

Comment:
This link will work from the `trait.MakeWriter.html` page at the create root (e.g. https://docs.rs/tracing-fmt/latest/tracing_fmt/trait.MakeWriter.html) but not from the `writer` sub-module page (e.g. https://docs.rs/tracing-fmt/latest/tracing_fmt/writer/trait.MakeWriter.html). Would you like me to use an absolute URL instead?

I also noticed a similar issue with `tracing::Subscriber`; compare the links to `Interest` on these two pages:

- https://docs.rs/tracing/0.1.6/tracing/trait.Subscriber.html
- https://docs.rs/tracing/0.1.6/tracing/subscriber/trait.Subscriber.html

Reply:
Hmm, yeah, we should probably fix that. 

My concern with absolute URLs is that if we use an absolute docs.rs URL, then we'll link to docs.rs even in local builds or (eventually) master RustDoc built on CI, which could be potentially confusing. It's possible that there's no solution that avoids this while also fixing the problem you mentioned though.